### PR TITLE
Fix, refine `garnish_variants` variant intersection

### PR DESCRIPTION
**Pull request by andrewrech**
* shorten documentation, correct
* fail if VCF is not SnpEff-annotated
* remove variant filtering for consistency, outside scope
* remove SnpEff annotation NAs inside parallelized loop
* merge on transcript ID and cDNA change
* keep informative INFO and FORMAT columns from all input VCFs
* do not prematurely bind dt

### CODE REVIEW

**Comment by andrewrech** [comment]
@leeprichman passes long tests

File: tests/testthat/test-garnish_variants_intersection.R [hunk 1]
```diff
@@ -14,10 +14,18 @@ testthat::test_that("garnish_variants intersection", {
...
+  testthat::expect_equal(dto_i %>% length, 23)
+  testthat::expect_equal(dto_i %>% nrow, 64)
+
+    # run test
+    dto_ni <- garnish_variants(c("antigen.garnish_example_mutect2.vcf",
+                     "antigen.garnish_example_strelka.vcf"),
+            intersect = FALSE)
+
+  testthat::expect_equal(dto_ni %>% length, 23)
+  testthat::expect_equal(dto_ni %>% nrow, 118)
```
Comment (leeprichman) [inline tests/testthat/test-garnish_variants_intersection.R]: I think ideally changes to union_vcf above will not affect this test, just the content of the table.  this table likely had NA's for REF, CHROM, POS etc. right?

File: R/antigen.garnish_io.R [hunk 2]
```diff
@@ -296,62 +290,83 @@ garnish_variants <- function(vcfs, intersect = TRUE){
     vdt %<>% .[!se %likef% "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS"]
     vdt %<>% .[!se %likef% "WARNING_TRANSCRIPT_NO_START_CODON"]
 
+    # filter out NA
+    vdt %<>% .[!ensembl_transcript_id %>% is.na &
+               !cDNA_change %>% is.na]
+
```
[the remark this answers is not in the document]
Comment (andrewrech) [reply R/antigen.garnish_io.R]: Don't think. To pass a variants needs to evaluate to `TRUE` on both columns, e.g. have both be not `NA`. If it were OR, a variant would pass even if one were `NA`, which is not what we want.

File: tests/testthat/test-garnish_variants_intersection.R (hunk 1 above)
Comment (andrewrech) [reply tests/testthat/test-garnish_variants_intersection.R]: Yes you are right. We actually want to subset to variants present in both and merge those, then `rbindlist` to variants present in either vcf.

Comment (andrewrech) [reply tests/testthat/test-garnish_variants_intersection.R]: Fixed

File: R/antigen.garnish_io.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewrech) [reply R/antigen.garnish_io.R]: No, both need to be ! `NA`

**Comment by andrewrech** [comment]
@leeprichman thanks. passes long test

File: R/antigen.garnish_io.R [hunk 3]
```diff
@@ -264,25 +259,30 @@ garnish_variants <- function(vcfs, intersect = TRUE){
...
                       paste(collapse = "_") %>%
                       stringr::str_replace("\\.bam", "")
-      # extract vcf type
-      vcf_type <- vcf@meta %>% unlist %>% stringr::str_extract(stringr::regex("Strelka|Mutect", ignore_case = TRUE)) %>% stats::na.omit %>%
-                  unlist %>% data.table::first
+  # extract vcf type
+      vcf_type <- vcf@meta %>%
+        unlist %>%
+        stringr::str_extract(stringr::regex("(Strelka)|(Mutect)|(VarScan)|(samtools mpileup)|(somaticsniper)|(freebayes)|(virmidr)",
+          ignore_case = TRUE)) %>%
```
Comment (leeprichman) [inline R/antigen.garnish_io.R]: why virmidr? typo?

**Pull request merged**